pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
 	}
 	apiScheduleRes.Policies = []mongodbatlas.Policy{apiPolicy}
 
-	currentSchedule, _, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
+	currentSchedule, response, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
 	if err != nil {
 		r.Log.Debugf("unable to delete current backup policy for project: %v:%v, %v", projectID, cName, err)
+		// Should we return error here?
 	}
 	if currentSchedule == nil {
-		r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
-		return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api: %v", currentSchedule)
+		if response != nil {
+			r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. err: %w, response: %v", err, response)
+		} else {
+			r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. err: %w", err)
```

Comment:
Well, I suggest we rewrite this section. At this stage the `error` is likely to be <nil> and it's actually better to log something from the response as long as it isn't empty. Response is an object but I think we can just log only status from it.

Reply:
Done. Thank you